### PR TITLE
Update gemomedian product names.

### PR DESCRIPTION
**Pull request by tebadi**

### CODE REVIEW

File: products/ga_s2_gm.odc-product.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: ga_s2_gm
+name: gm_s2_annual
 description: Annual Geometric Median, Sentinel-2
```
Comment (alexgleith) [inline products/ga_s2_gm.odc-product.yaml:2]: Can we please update this too

File: products/ga_s2_clear_pixel_count.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: ga_s2_clear_pixel_count
+name: pc_s2_annual
 description: Clear Pixel Count S2A/B
```
Comment (alexgleith) [inline products/ga_s2_clear_pixel_count.yml:2]: And this, please!

**Review by alexgleith — CHANGES_REQUESTED** [review]
I'd like us to add a license section too, see: 

https://raw.githubusercontent.com/GeoscienceAustralia/digitalearthau/develop/digitalearthau/config/eo3/products-aws/ga_ls_wo_3.odc-product.yaml

File: products/ga_s2_clear_pixel_count.yml (hunk 2 above)
Comment (Kirill888) [reply products/ga_s2_clear_pixel_count.yml:2]: Probably best to rename yaml file as well to match the new name.

Comment (alexgleith) [reply products/ga_s2_clear_pixel_count.yml:2]: Great idea. Also, keep it to the new convention `<product_name>.odc-product.yaml`

File: products/ga_s2_gm.odc-product.yaml (hunk 1 above)
Comment (tebadi) [reply products/ga_s2_gm.odc-product.yaml:2]: Done.

File: products/ga_s2_clear_pixel_count.yml (hunk 2 above)
Comment (tebadi) [reply products/ga_s2_clear_pixel_count.yml:2]: Done.

File: products/ga_s2_gm.odc-product.yaml (hunk 1 above)
Comment (alexgleith) [reply products/ga_s2_gm.odc-product.yaml:2]: This should be something like:

> Surface Reflectance Annual Geometric Median and Absolute Deviations, Sentinel-2

It's not just the geomedian anymore.

File: products/ga_s2_clear_pixel_count.yml (hunk 2 above)
Comment (alexgleith) [reply products/ga_s2_clear_pixel_count.yml:2]: Can we update the description to align with the geomedian product. Same structure :-)

File: products/gm_s2_annual.odc-product.yaml [hunk 3]
```diff
@@ -1,10 +1,10 @@
-name: ga_s2_gm
-description: Annual Geometric Median, Sentinel-2
+name: gm_s2_annual
+description: Surface Reflectance Annual Geometric Median Absolute Deviations (Cosine/Euclidean/Bray-Curtis distance), Sentinel-2
```
Comment (alexgleith) [inline products/gm_s2_annual.odc-product.yaml:2]:
```suggestion
description: Surface Reflectance Annual Geometric Median and Median Absolute Deviations, Sentinel-2
```

**Comment by alexgleith** [comment]
> I'd like us to add a license section too, see:
> 
> https://raw.githubusercontent.com/GeoscienceAustralia/digitalearthau/develop/digitalearthau/config/eo3/products-aws/ga_ls_wo_3.odc-product.yaml

Hey @tebadi can you please add a license field to both the files :-)

**Review by alexgleith — APPROVED** [review]

**Pull request merged**